interface-definitions/containers.xml.in

```diff
@@ -64,6 +64,75 @@
               </leafNode>
             </children>
           </tagNode>
+          <tagNode name="port">
+            <properties>
+              <help>Publish port to the container</help>
+            </properties>
+            <children>
+              <leafNode name="source">
+                <properties>
+                  <help>Source host port</help>
+                  <valueHelp>
+                    <format>u32:1-65535</format>
+                    <description>Source host port</description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>start-end</format>
+                    <description>Source host port range (e.g. 10025-10030)</description>
+                  </valueHelp>
+                </properties>
+              </leafNode>
+              <leafNode name="destination">
+                <properties>
+                  <help>Destination container port</help>
+                  <valueHelp>
+                    <format>u32:1-65535</format>
+                    <description>Destination container port</description>
+                  </valueHelp>
+                  <valueHelp>
+                    <format>start-end</format>
+                    <description>Destination container port range (e.g. 10025-10030)</description>
+                  </valueHelp>
+                </properties>
+              </leafNode>
+              <leafNode name="protocol">
+                <properties>
+                  <help>Protocol tcp/udp</help>
+                  <completionHelp>
+                    <list>tcp udp</list>
+                  </completionHelp>
+                  <constraint>
+                    <regex>^(tcp|udp)$</regex>
+                  </constraint>
```

Comment:
What will be the default if none is specified? Can I specify both?

Reply:
You can specify both with different map
```
set container name foo image 'busybox'
set container name foo network NET01
set container name foo port map destination '22'
set container name foo port map source '2222'
set container name foo port map protocol udp
set container name foo port map2 destination '22'
set container name foo port map2 source '2222'
set container name foo port map2 protocol tcp
set container network NET01 prefix '192.0.2.0/24'
```
Check
```
vyos@r6-roll# run show container 
CONTAINER ID  IMAGE                             COMMAND  CREATED        STATUS            PORTS                                       NAMES
72d6e03c9a04  docker.io/library/busybox:latest  sh       6 seconds ago  Up 6 seconds ago  0.0.0.0:2222->22/tcp, 0.0.0.0:2222->22/udp  foo

```